test/jdk/tools/lib/tests/JImageValidator.java

```diff
@@ -225,10 +222,6 @@ public long getModuleLauncherExecutionTime() {
     }
 
     public static void readClass(byte[] clazz) throws IOException {
-        try (InputStream stream = new ByteArrayInputStream(clazz)) {
-            ClassFile.read(stream);
-        } catch (ConstantPoolException e) {
-            throw new IOException(e);
-        }
+        Classfile.of().parse(clazz);
```

Comment:
Notice that the old API might throw different exceptions compared to the new one; the new one is lazy, e.g. if there's broken interface entry but the interfaces are not queried, no exception is thrown.

Reply:
You are right, however original `ClassFile::read` also does not represent much of a validation.
Stronger verification can by applied by asserting no verify errors returned from `ClassModel::verify`.